Added Startup Script which installs Python  + PreInstall window with mirrorlist creation + Disk Parting + Mounting + Filesystems + Pacstrap

mame98 opened the pull request:
Large Changes:
- Added Startup script:
  - Checks for Python if `python ArchSetup` fails
  - If installer fails, the script tells the user to use a stable version and or to report the bug
- PreInstall Window
  - Installs Refelctor (Tool to generate mirrorlists)
  - Generates Mirrorlists (`reflector --verbose -l 50 -p http --sort rate --save list.txt`) which gets the 50 most up to date servers and sort them by download speed
  - Partitioning (sdX1: datapart (size: DISKSIZE - RAM) ; sdX2: swap (size: RAM))
  - Create Filesystem + SWAP
  - Mounting
  - Provides feedback to the user of the current status
- Install Window
  - Running pacstrap and printing the most recent line of it's output to the Install window, so the user can track the progress (I was unable to create a scrolling output :( but I think a updated single line is enought for now...)
  - Installing the Groups base + base-devel
  
  Minor Changes:
  - TextWidget will now recalculate it's size if the text is updated
  - Added a append function to TextWidget, but it is currently only a `pass`
  - Draw function of TextWidget contains a try/except block around the addstr code, because a large output will might go offscreen and causes an exception... However except does nothing (`pass`)...

Sorry for the many commits, but I tested it with QEMU so I had to commit every single modification and download it again on the Install ISO ;-)

I will now start working on the Configuration part (`arch-chroot`) I think we should use GRUB by default if you have another opinion just tell me...

mame98 committed:
Added startup script

mame98 committed:
Added packagelist update to setup.sh

mame98 committed:
Improved Label on WelcomeWindow
	-> Added informations about [SPACE]

mame98 committed:
Added PreInstall + new event 'showed' (after resizing) + TextWidget is now modifyable

.gitignore updated to ignore list.txt (mirrorlist)

mame98 committed:
Added missing translations

mame98 committed:
Added installing of requiered software

mame98 committed:
Added missing translations

mame98 committed:
Added unfunctional Prototype of DiskPartWindow

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Removed useless function

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Fixed Typo...

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Fixed bug in text on DiskPartWindow

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
First basic testing implementation of the disk formarter

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Partitioning should now be dynamic + added *.pyc to .gitignore

Because I accidently call python instead of python3 I dont want to spam the repo ;-)

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Fixed Partition Bugs

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Forgot /dev/ in Disks().part_disk

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Added first filessystem implementation

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Added Force override + Removed output from screen

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Added Mounting

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Removed Warnings

It is basicly working right now

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Added Installwindow + Pacstrap + append line option for TextWidget

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
First atempt to fix InstallWindow

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Bugfix Test

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Now Install window could work ;-)

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Fixed type Staus to Status

it is now working, but the user gets no notifications + app freezes + prints weird messages ;-)

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Improve Output

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Improved output even further

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Improved append function


Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Various Bugs fixed + removed append function

it must be rewritten soon...

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
pacstrap output is now decoded as UTF-8 + removed scroll widget

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Added resizing if text of TextWidget is changed + Workaround for crash

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
added translations

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Minor Interface Improvements

-  Removed SpacerWidget on InstallWindow
-  Changed TextWidgets Location from (3,1) to (4,1)

Should now be stable

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

antoniovazquezblanco commented:
WOW! Very nice!

I personally use EFI and I think that it will be the way fordward. I would like to add an option compatible with that although most desktops aren't compatible with that. Anything you decide will be very welcome :).

I am unable to test the changes now but I will merge.

Thanks! ;)

mame98 committed:
First PostInstall implementations

* fstab
* Hostname
* Timezone
* Keyboard Layout
* Console Font

mame98 committed:
Implemented some more PostInstall things

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Some BuxFixes in PostInstall

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Fixed Typo + Wrong file paths in PostInstall

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Implemented Mirrorlist Installation + DB upgrade

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Fix Passwd setting

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
New Window (MainLocaleWindow) + locale.conf in PostInstall

The window will ask the user to choose its main locale, so the locale he/she uses by default

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Minor Bux Fix (added LANG=)

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Some Bug Fixing with MainLocales

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Added MainLocaleWindow to window list

...

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Imported wrong Widget

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

mame98 committed:
Fixed MainLocaleWindow

mame98 committed:
Implemented first version of a /etc/locale.conf writer

(UNTESTED)

mame98 committed:
Merge pull request #7 from jetspace/dev

Dev

mame98 committed:
Fixed Locale.gen writing

Signed-off-by: Marius Messerschmidt <marius.messerschmidt@googlemail.com>

antoniovazquezblanco commented:
Sorry I didn't merge! I was looking for the changes to test and realized I clicked on the wrong button!
